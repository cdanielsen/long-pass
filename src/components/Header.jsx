import React from 'react'
import PropTypes from 'prop-types'

function Header (props) {
  return (
    <div
      id={props.id}>
      <h1>{props.textContent}</h1>
    </div>
  )
}

Header.propTypes = {
  textContent: PropTypes.string,
  id: PropTypes.string
}

export default Header
