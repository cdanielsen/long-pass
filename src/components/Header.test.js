import React from 'react'
import Header from './Header.jsx'
import { shallow, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-15'

configure({adapter: new Adapter()})

describe('Header component', () => {
  test('Should return a div with the correct class and text', () => {
    const className = 'stuff'
    const textContent = 'Yolo!'
    const wrapper = shallow(<Header className={className} textContent={textContent} />)

    expect(wrapper).toMatchSnapshot()
    expect(wrapper.prop('className')).toEqual(className)
    expect(wrapper.text()).toEqual(textContent)
  })
})
