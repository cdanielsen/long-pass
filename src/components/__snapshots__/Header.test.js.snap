// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Should return a div with the correct class and text 1`] = `
<div
  className="stuff"
>
  Yolo!
</div>
`;
